Fix npm scripts

neilff opened the pull request:
- Added `npm run dev`
- Changed `npm run start` to clean, build, and run simple express server

https://github.com/rangle/react-redux-seed/issues/11

SethDavenport commented on server.js:
```diff
@@ -0,0 +1,18 @@
+const PORT = process.env.PORT || 3000;
+
+app.get('*', function(req, res) {
+  res.sendFile(path.join(__dirname, 'index.html'));
```
How would this handle asset files?  Or are we bundling all the things (images etc) with webpack?

neilff commented:
@SethDavenport I've modified the PR quite a bit, I added a prod and dev config for webpack. I've added a simple asset as an example, the webpack `url-loader` will move these assets over to `/dist` in production. I modified the server to use `express.static` on the `dist` folder, and instead only `GET /` will return `index.html`

SethDavenport commented on webpack.config.prod.js:
```diff
@@ -0,0 +1,65 @@
+    publicPath: '/dist/'
+  },
+  plugins: [
+    new webpack.optimize.OccurenceOrderPlugin(),
```
Is this a typo or is the plugin actually called that?

neilff replied:
The `OccuranceOrderPlugin` is real, https://webpack.github.io/docs/list-of-plugins.html#occurenceorderplugin

SethDavenport commented on webpack.config.prod.js:
```diff
@@ -0,0 +1,65 @@
+var path = require('path');
```
Was discussing with the possibility of sharing one webpack config between dev and prod.  I know you can control minification for example using a command line arg to webpack.  Anyway something to think about.

neilff replied:
My only concern is when the files start to diverge, this can become a nightmare to read through the `if` `else` to determine what is going on. However, right now this file is simple enough, I'll merge the two and when it becomes an issue down the road we can look into refactor.

andrejkn commented:
@neilff looks good :shipit:

neilff commented:
@andrejkn I'll combine the two `dev` / `prod` webpack files shortly, and then have you take a look once more, please don't merge until then
